templates/quickstart-database-for-atlassian-services.yaml

```diff
@@ -170,10 +179,16 @@ Resources:
     Type: AWS::CloudFormation::Stack
     Condition: UseAurora
     Properties:
-      TemplateURL:
-        !Sub
-        - https://${QSS3BucketName}.${QSS3Region}.amazonaws.com/${QSS3KeyPrefix}submodules/quickstart-atlassian-services/submodules/quickstart-amazon-aurora/templates/aurora_postgres.template.yaml
-        - QSS3Region: !If ["GovCloudCondition", "s3-us-gov-west-1", "s3"]
+      TemplateURL: !If
+        # Maintain backward compatibility for non-region support
+        - NoBucketRegionSupplied
+        - !Sub
+          - https://${QSS3BucketName}.${S3Region}.${AWS::URLSuffix}/${QSS3KeyPrefix}submodules/quickstart-atlassian-services/submodules/quickstart-amazon-aurora/templates/aurora_postgres.template.yaml
+          - S3Region: !If ["GovCloudCondition", "s3-us-gov-west-1", "s3"]
+        - !Sub
+          - https://${S3Bucket}.s3.${S3Region}.${AWS::URLSuffix}/${QSS3KeyPrefix}submodules/quickstart-atlassian-services/submodules/quickstart-amazon-aurora/templates/aurora_postgres.template.yaml
+          - S3Bucket: !If [UsingDefaultBucket, !Sub '${QSS3BucketName}-${AWS::Region}', !Ref QSS3BucketName]
+            S3Region: !If [UsingDefaultBucket, !Ref 'AWS::Region', !Ref QSS3BucketRegion]
```

Comment:
Historically this template has never been supplied with the parameters; QSS3BucketName or QSS3KeyPrefix therefore it has ALWAYS been interpolated (regardless of product stack) to its default value which in this case is:
http://aws-quickstart.s3.amazonaws.com/quickstart-atlassian-services/templates/quickstart-bastion-for-atlassian-services.yaml

This is also the case for templates sourced from custom S3 buckets! If we change this template we need to make sure it will accommodate those product templates which are being called from custom S3 buckets and do not pass the parameters; QSS3BucketName, QSS3KeyPrefix. 
             
If we dont do this stack deployments will fail! 
             
As a result we support both scenarios, the first condition, if met, will construct a non region specific URL for those templates that do not provide the mentioned parameters. The second condition will construct a region specific URL for those templates that do.